packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -107,10 +119,57 @@ class DDSCarousel extends LitElement {
         // @ts-ignore
         this._observerResizeContainer = new ResizeObserver(this._observeResizeContainer);
         this._observerResizeContainer.observe(contentsNode);
+        containerNode?.addEventListener('scroll', this._handleScrollFocus);
       }
     }
   }
 
+  /**
+   * Stops the container from scrolling when focusing on a card outside of the viewport.
+   *
+   * @param event The event.
+   */
+  // eslint-disable-next-line class-methods-use-this
+  private _handleScrollFocus({ target }: Event) {
+    (target as HTMLElement).scrollTo(0, 0);
+  }
+
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldNotContains = target !== this && !this.contains(relatedTarget as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // keep current page if tabbing back into the carousel after previously moving pages
+    if (currentContains && oldNotContains) {
+      // focus coming from the top
+      if (currentCardIndex === 0) {
+        (this.children[this.start] as HTMLElement).focus();
+
+        // focus coming from bottom
+      } else if (currentCardIndex === this._total - 1) {
+        (this.children[this.start + this.pageSize - 1] as HTMLElement).focus();
+      }
+      return;
+    }
```

Comment:
Could you elaborate why you think we still need this logic here? I thought the first card automatically gets focus in forward-tab scenario, and last card automatically gets on focus in backward-tab scenario. If it's not the case, could you point out what element gets focus instead in such scenarios?

Reply:
Sure, we still need this logic because the page will not automatically focus on the first/last page. The focus actually either goes to the very first card (index 0), or to the very last one, so even if the current page is the right one, the focus will be entirely out of view, and the user would need to keep tabbing/shift+tabbing until it reaches the current page if we don't have the `focus()` here.

Reply:
Thank you for clarifying. My thought is tabbing on the very first card, instead of the first card on the current page, in forward-tab scenario. Similar for backward-scenario. Could you work with the designer to double-check that? Thanks!

Reply:
@asudoh Discussed with @photodow and confirmed that focusing back on the current page (as it is currently) works best.

Reply:
Thanks @IgnacioBecerra - Please add a comment as such.